src/modules/cmdpal/Microsoft.CmdPal.UI/Controls/ContextMenu.xaml.cs

```diff
@@ -0,0 +1,223 @@
+    {
+        if (e.Handled)
+        {
+            return;
+        }
+
+        var ctrlPressed = InputKeyboardSource.GetKeyStateForCurrentThread(VirtualKey.Control).HasFlag(CoreVirtualKeyStates.Down);
+        var altPressed = InputKeyboardSource.GetKeyStateForCurrentThread(VirtualKey.Menu).HasFlag(CoreVirtualKeyStates.Down);
+        var shiftPressed = InputKeyboardSource.GetKeyStateForCurrentThread(VirtualKey.Shift).HasFlag(CoreVirtualKeyStates.Down);
+        var winPressed = InputKeyboardSource.GetKeyStateForCurrentThread(VirtualKey.LeftWindows).HasFlag(CoreVirtualKeyStates.Down) ||
+            InputKeyboardSource.GetKeyStateForCurrentThread(VirtualKey.RightWindows).HasFlag(CoreVirtualKeyStates.Down);
+
+        var result = ViewModel?.CheckKeybinding(ctrlPressed, altPressed, shiftPressed, winPressed, e.Key);
+
+        if (result == ContextKeybindingResult.Hide)
+        {
+            e.Handled = true;
+            WeakReferenceMessenger.Default.Send<CloseContextMenuMessage>();
+            UpdateUiForStackChange();
+        }
+        else if (result == ContextKeybindingResult.KeepOpen)
+        {
+            e.Handled = true;
+        }
+        else if (result == ContextKeybindingResult.Unhandled)
+        {
+            e.Handled = false;
+        }
+    }
+
+    private void ViewModel_PropertyChanged(object? sender, System.ComponentModel.PropertyChangedEventArgs e)
+    {
+        var prop = e.PropertyName;
+
+        if (prop == nameof(ContextMenuViewModel.FilteredItems))
+        {
+            UpdateUiForStackChange();
+        }
+    }
+
+    private void ContextFilterBox_TextChanged(object sender, TextChangedEventArgs e)
+    {
+        ViewModel?.SetSearchText(ContextFilterBox.Text);
+
+        if (CommandsDropdown.SelectedIndex == -1)
+        {
+            CommandsDropdown.SelectedIndex = 0;
+        }
+    }
+
+    private void ContextFilterBox_KeyDown(object sender, KeyRoutedEventArgs e)
+    {
+        var ctrlPressed = InputKeyboardSource.GetKeyStateForCurrentThread(VirtualKey.Control).HasFlag(CoreVirtualKeyStates.Down);
+        var altPressed = InputKeyboardSource.GetKeyStateForCurrentThread(VirtualKey.Menu).HasFlag(CoreVirtualKeyStates.Down);
+        var shiftPressed = InputKeyboardSource.GetKeyStateForCurrentThread(VirtualKey.Shift).HasFlag(CoreVirtualKeyStates.Down);
+        var winPressed = InputKeyboardSource.GetKeyStateForCurrentThread(VirtualKey.LeftWindows).HasFlag(CoreVirtualKeyStates.Down) ||
+            InputKeyboardSource.GetKeyStateForCurrentThread(VirtualKey.RightWindows).HasFlag(CoreVirtualKeyStates.Down);
+
+        if (e.Key == VirtualKey.Enter)
+        {
+            if (CommandsDropdown.SelectedItem is CommandContextItemViewModel item)
+            {
+                if (InvokeCommand(item) == ContextKeybindingResult.Hide)
+                {
+                    WeakReferenceMessenger.Default.Send<CloseContextMenuMessage>();
+                }
+
+                UpdateUiForStackChange();
```

Comment:
⛔
This was originally 

```cs
                if (ViewModel?.InvokeItem(item) == ContextKeybindingResult.Hide)
                {
                    MoreCommandsButton.Flyout.Hide();
                    WeakReferenceMessenger.Default.Send<FocusSearchBoxMessage>();
                }
                else
                {
                    UpdateUiForStackChange();
                }
```

and I think that's why when you click on "We can go deeper..." it both sends the StatusMessage, and opens the context menu to its `MoreCommands`

Reply:
Yup. My thought was "if you click on item X and it has a command, we should execute that command." If it also has "MoreCommands" we should then display the nested menu, otherwise hide the menu. 

Just to confirm, you'd want to not fire item X's command, but instead list item X at the top of the nested commands?